e2e_playwright/st_text_input_test.py

```diff
@@ -146,6 +146,35 @@ def test_text_input_has_correct_value_on_click_outside(app: Page):
     )
 
 
+def test_text_input_does_not_trigger_rerun_when_value_does_not_change_and_click_outside(
+    app: Page,
+):
+    """Test that st.text_input has the correct value on click outside."""
+
+    first_text_input_field = (
+        app.get_by_test_id("stTextInput").first.locator("input").first
+    )
+    first_text_input_field.focus()
+    first_text_input_field.fill("hello world")
+    app.get_by_test_id("stMarkdown").first.click()
+
+    expect(
```

Comment:
nit: this could also use the `expect_markdown` in app_utils

Reply:
btw I think it needs to be `Rerun counter: 0` or you need to start with 1 in the script

Reply:
Thanks for the hint, but actually the counter is incremented before the write 🙂

Reply:
>nit: this could also use the expect_markdown in app_utils

`expect_markdown` calls `to_be_visible()` whereas here we call `to_be_attached`, since the counter is on the bottom of the app, we would need to scroll to it first and to do that, I guess we would need to call `app.get_by_test_id("stMarkdown").filter` at least once anyways. So I think it might not be worth it to swap the calls. We could think about adding an additional argument to `expect_markdown` though to swap between `to_be_visible` and `to_be_attached`.